### ISSUE TITLE
Can't mount using mgmt into a folder created by mgmt

### ISSUE THREAD

**Issue by wdullaer** [purpleidea/mgmt#504]
## Versions:

* mgmt version (eg: `mgmt --version`): 0.0.18

* operating system/distribution (eg: `uname -a`): Ubuntu 19.04

## Description:
This issue is surprisingly similar to #503 : You can't mount a disk into a folder you create in the same mgmt run.

This will result in the following error:
```
20:59:11 main.go:496: gapi: generating new graph...
20:59:11 main.go:496: gapi: lang: running interpret...
20:59:11 main.go:496: gapi: lang: funcs: func `str(noatime)` stopped
20:59:11 main.go:496: gapi: lang: funcs: func `str()` stopped
20:59:11 main.go:496: gapi: lang: funcs: func `str()` stopped
20:59:11 main.go:496: gapi: lang: funcs: func `map(str(noatime): str(), str(defaults): str())` stopped
20:59:11 main.go:192: main: error validating the new graph: no such file or directory
error validating mount point: /mnt/disk1
```

https://github.com/purpleidea/mgmt/blob/d5b7dc0acc3b608348c891a3f274f6ee3bad302d/engine/resources/mount.go#L153

The Validate function of the mount resource will try to check the existance of the folder. This function is called when parsing the input, long before the resource graph is executed. This means it's not aware of the changes mgmt might be making.

I think in general Validate should contain "static analysis" checks and any checks which involve the runtime state of the machine should happen when the resource is applying its state.

**Comment by purpleidea** [comment]
Agreed. Can you post your `mcl` code too please?

**Comment by wdullaer** [comment]
Here you go

```
file "/mnt/disk1/" {
	state => "exists",
}

mount "/mnt/disk1" {
	state => "exists",
	device => "UUID=8f3b7486-7529-4552-820c-284232c8354d",
	type => "ext4",
	options => {
		"noatime"=>"",
		"defaults"=>"",
	},
}
```

**Comment by purpleidea** [comment]
> I think in general Validate should contain "static analysis" checks and any checks which involve the runtime state of the machine should happen when the resource is applying its state.

This is completely true. We may have gotten a little overzealous in putting checks in Validate that really should have happened later. As a result of you being an early user, you're helping shake out these issues, and I really appreciate that, thanks!
